sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetUtils.scala

```diff
@@ -144,6 +145,30 @@ object ParquetUtils {
       file.getName == ParquetFileWriter.PARQUET_METADATA_FILE
   }
 
+  /**
+   * Whether columnar read is supported for the input `schema`.
+   */
+  def isBatchReadSupportedForSchema(sqlConf: SQLConf, schema: StructType): Boolean =
```

Comment:
It would be great to design new functions minimally which receive the minimal information instead of passing over the full `SQLConf` recursively. For example, could you revise
- `isBatchReadSupportedForSchema` by handing over the values of `sqlConf.parquetVectorizedReaderEnabled` and `sqlConf.parquetVectorizedReaderNestedColumnEnabled`
- `isBatchReadSupported` by handing over the value of `sqlConf.parquetVectorizedReaderNestedColumnEnabled` from `isBatchReadSupportedForSchema`.

directly instead of `SQLConf` recursively?

Reply:
Makes sense. Let me change it.